matrix_is_tester/base_api_test.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+import unittest
+
+# These are standard python unit tests, but are generally intended
+# to be run with trial. Trial doesn't capture logging nicely if you
+# use python 'logging': it only works if you use Twisted's own.
+from twisted.python import log
+
+from .is_api import IsApi
+from .launch_is import get_or_launch_is
+from .mailsink import get_shared_mailsink
+
+
+# Not a test case itself, but can be subclassed to test APIs common
+# between versions. Subclasses provide self.api
+class BaseApiTest:
+    def setUp(self):
+        self.baseUrl = get_or_launch_is()
+
+        self.mailSink = get_shared_mailsink()
+
+        self.api = IsApi(self.baseUrl, self.API_VERSION, self.mailSink)
+
+        random.seed(1)
```

Comment:
why do we do this?

Reply:
It's not a pattern I'm familiar with. Also, it feels like the `random.seed`s are scattered sligthly arbitrarily?

Reply:
mm, yeah - does suffer from there not being a single place to put init code. Maybe it's more confusing than it is useful as I don't think it will matter in practice.